### PR TITLE
add sign in button to Code Studio homepage, mobile

### PR DESCRIPTION
**Pull request by tanyaparker**
**iPhone 6**
image

**Android**
image

**iPad 2**
image

### CODE REVIEW

File: dashboard/app/assets/stylesheets/application.css.scss [hunk 1]
```diff
@@ -215,6 +215,16 @@ img.video_thumbnail {
   }
 }
 
+.sign_in.show_only_on_mobile {
+  float: right;
+  margin-right: 10px;
+  margin-bottom: 10px;
+
+  @media screen and (max-device-height: 900px) and (max-device-width: 500px) {
```
Comment (breville) [inline dashboard/app/assets/stylesheets/application.css.scss]: do we use max-device-height anywhere else?  i thought our mobile layout was keyed entirely off of width?

**Comment by breville** [comment]
should the right edge be aligned with the right edge of the purple header?

at any rate, lgtm

File: dashboard/app/assets/stylesheets/application.css.scss (hunk 1 above)
Comment (tanyaparker) [inline dashboard/app/assets/stylesheets/application.css.scss]: Yeah. I decided to use the same conditions that we use for hiding the header, just so they would never appear both at the same time.

Comment (tanyaparker) [inline dashboard/app/assets/stylesheets/application.css.scss]: It probably doesn't matter. I could remove height from both of them?

**Comment by tanyaparker** [comment]
Yeah I can't decide which I like better when aligning next to super rounded corners. I sort of prefer not aligning to the right because it creates this weird pocket of white space. But I'm not tied to it strongly, I can switch it back to same right alignment.

image

image

File: dashboard/app/assets/stylesheets/application.css.scss (hunk 1 above)
Comment (breville) [inline dashboard/app/assets/stylesheets/application.css.scss]: That's interesting.  You could, but only if you want to test that it works in all the right places.  If you want the more surgical fix, then matching the criteria sounds like the right thing to do.

Btw, have you tested at exactly 500px and 900px?  Sometimes we need to be off-by-one, depending whether the ranges are inclusive or exclusive.

**Comment by breville** [comment]
To me the second one feels more right, despite the pocket.  Otherwise it makes it look like Sign In is sitting on top of the flat part of the purple and risks sliding off if it goes further to the right :)

**Comment by tanyaparker** [comment]
alright alright

image

**Comment by tanyaparker** [comment]
Closing because staging-next is no longer needed. Merging into staging https://github.com/code-dot-org/code-dot-org/pull/9088

**Pull request closed without merging**